### ISSUE TITLE
What is Visual Studio support like?

### ISSUE THREAD

**Issue by haikarainen** [andrewrk/libsoundio#175]
I could only ever find window compilation instructions for mingw. Is there any support for Visual Studio 2017 (or even 2015)?

What would be the most straight forward way to compile this library, under windows, with the VS toolchain?

**Comment by tresf** [comment]
@haikarainen, often it's something like this:

1. Install CMake for Windows

2. Build from CMD
    ```cmd
    git clone https://github.com/andrewrk/libsoundio
    cd libsoundio
    mkdir build
    cmake ..
    cmake --build .
    ```

3. Unfortunately this won't compile on my machine.  Perhaps I'm missing something obvious like `-G "Visual Studio 15 2017"`?

**Comment by haikarainen** [comment]
@tresf  I agree, this is what it _should_ be like. Sadly, it seems like Windows is a second-hand citizen in libsoundio :/

**Comment by mdsitton** [comment]
I once had it building on windows, but I've since switched to using cubeb. I've been considering trying soundio again though so we'll see how it goes.

**Comment by tresf** [comment]
> @tresf I agree, this is what it should be like. Sadly, it seems like Windows is a second-hand citizen in libsoundio :/

Probably not the best way to phrase "Please add Visual Studio support".  These projects are open source for a reason.  If you take a look through the pull requests, many of them are crowdsourced.  Please consider telling the project how to work versus making bold statements like "Sadly" and "Second [class] citizen".  Figure it out, PR it and make the world a better place.  That's what GitHub is about. :)

**Comment by tzaeru** [comment]
I actually got the library itself to compile. I just did
```
mkdir build
cd build
cmake ..
cmake --build .
```
Nothing more. I had Visual Studio 2017 and Visual Studio 2017 CMake tools (you can install them with the VS2017 installer, by going to components. They are.. Somewhere there..) installed. After that, I had soundio.dll and unit_tests.exe under the Debug directory.

Some things don't build. For example, the underflow and backend_disconnect_recover tests try to include `unitstd.h` which is specific to POSIX systems. The record example uses __attribute__, a macro specific to GCC.

I'll see if I can get it to compile a bit more smoothly with VS..

**Comment by conradjones** [comment]
Builds fine for me under windows / MSVC. I use the build tools only not full visual studio as i develop on linux and let the CI system do the windows and mac builds.

effectively what I'm doing is as above running cmake, what I'm actually doing is getting jenkins to trigger conan to trigger cmake,
